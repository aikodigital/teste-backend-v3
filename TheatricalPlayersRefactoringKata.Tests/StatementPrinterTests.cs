using System;
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using Xunit;

namespace TheatricalPlayersRefactoringKata.Tests;

public class StatementPrinterTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TestStatementExampleLegacy()
    {
        var plays = new Dictionary<string, Play>();
        plays.Add("hamlet", new Play("Hamlet", 4024, "tragedy"));
        plays.Add("as-like", new Play("As You Like It", 2670, "comedy"));
        plays.Add("othello", new Play("Othello", 3560, "tragedy"));

        Invoice invoice = new Invoice(
            "BigCo",
            new List<Performance>
            {
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40),
            }
        );

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.Print(invoice, plays);

        Approvals.Verify(result);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TestTextStatementExample()
    {
        var plays = new Dictionary<string, Play>();
        plays.Add("hamlet", new Play("Hamlet", 4024, "tragedy"));
        plays.Add("as-like", new Play("As You Like It", 2670, "comedy"));
        plays.Add("othello", new Play("Othello", 3560, "tragedy"));
        plays.Add("henry-v", new Play("Henry V", 3227, "history"));
        plays.Add("john", new Play("King John", 2648, "history"));
        plays.Add("richard-iii", new Play("Richard III", 3718, "history"));

        Invoice invoice = new Invoice(
            "BigCo",
            new List<Performance>
            {
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40),
                new Performance("henry-v", 20),
                new Performance("john", 39),
                new Performance("henry-v", 20)
            }
        );

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.Print(invoice, plays);

        Approvals.Verify(result);
    }
}
