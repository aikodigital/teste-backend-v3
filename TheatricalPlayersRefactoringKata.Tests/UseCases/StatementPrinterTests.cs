using System;
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using TheatricalPlayersRefactoringKata.Application;
using TheatricalPlayersRefactoringKata.Domain;
using TheatricalPlayersRefactoringKata.Domain.Enums;
using Xunit;

namespace TheatricalPlayersRefactoringKata.Tests.UseCases;

public class StatementPrinterTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TestStatementExampleLegacy()
    {
        var hamletPlay = new Play("Hamlet", 4024, PlayTypeEnum.Tragedy);
        var asYLikePlay = new Play("As You Like It", 2670, PlayTypeEnum.Comedy);
        var othelloPlay = new Play("Othello", 3560, PlayTypeEnum.Tragedy);

        Invoice invoice = new Invoice(
            "BigCo",
            new List<Performance>
            {
                new Performance(hamletPlay, 55),
                new Performance(asYLikePlay, 35),
                new Performance(othelloPlay, 40),
            }
        );

        var statementPrinter = new StatementPrinter();
        var result = statementPrinter.Print(invoice);

        Approvals.Verify(result);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TestTextStatementExample()
    {
        var hamletPlay = new Play("Hamlet", 4024, PlayTypeEnum.Tragedy);
        var AsYLikeItPlay = new Play("As You Like It", 2670, PlayTypeEnum.Comedy);
        var othelloPlay = new Play("Othello", 3560, PlayTypeEnum.Tragedy);
        var henryVPlay = new Play("Henry V", 3227, PlayTypeEnum.History);
        var kingJohnPlay = new Play("King John", 2648, PlayTypeEnum.History);
        var richardIIIPlay = new Play("Richard III", 3718, PlayTypeEnum.History);

        Invoice invoice = new Invoice(
            "BigCo",
            new List<Performance>
            {
                new Performance(hamletPlay, 55),
                new Performance(AsYLikeItPlay, 35),
                new Performance(othelloPlay, 40),
                new Performance(henryVPlay, 20),
                new Performance(kingJohnPlay, 39),
                new Performance(henryVPlay, 20)
            }
        );

        var statementPrinter = new StatementPrinter();
        var result = statementPrinter.Print(invoice);

        Approvals.Verify(result);
    }

    [Fact]
    [UseReporter(typeof(DiffReporter))]
    public void TesteTextStatmentNoPerformanceExample()
    {
        Invoice invoice = new Invoice("BigCo", new List<Performance>());

        var statementPrinter = new StatementPrinter();

        var result = statementPrinter.Print(invoice);

        Approvals.Verify(result);
    }
}
