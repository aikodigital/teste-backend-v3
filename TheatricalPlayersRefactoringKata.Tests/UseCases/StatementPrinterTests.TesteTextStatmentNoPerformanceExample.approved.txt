Statement for BigCo
Amount owed is $0.00
You earned 0 credits
